addons/mail/static/src/js/systray.js

```diff
@@ -138,14 +125,104 @@ var MessagingMenu = Widget.extend({
     },
     on_click_channel: function (event) {
         var channel_id = $(event.currentTarget).data('channel_id');
-        var channel = chat_manager.get_channel(channel_id);
-        if (channel) {
-            chat_manager.open_channel(channel);
+        if (channel_id == 'channel_inbox') {
+            var message_id = $(event.currentTarget).data('message_id');
+            framework.redirect('mail/view?message_id='+message_id);
+        } else {
+            var channel = chat_manager.get_channel(channel_id);
+            if (channel) {
+                chat_manager.open_channel(channel);
+            }
+        }
+    },
+});
+
+/**
+ * Menu item appended in the systray part of the navbar, redirects to the next activities of all app
+ */
+var ActivityMenu = Widget.extend({
+    template:'mail.chat.ActivityMenu',
+    events: {
+        "click": "on_activity_menu_click",
+        "click .o_activity_filter_button, .o_mail_channel_preview": "on_activity_filter_click",
+    },
+    start: function () {
+        this.$activities_preview = this.$('.o_mail_navbar_dropdown_channels');
+        chat_manager.bus.on("activity_updated", this, this.update_counter);
+        chat_manager.is_ready.then(this.update_counter.bind(this));
+        this.update_activity_preview();
+        this.is_counter_updated = false;
+        return this._super();
+    },
+    is_open: function () {
+        return this.$el.hasClass('open');
+    },
+    update_counter: function (data) {
+        if (data) {
+            if (data.activity_created) {
+                this.activity_counter ++;
+            }
+            if (data.activity_deleted && this.activity_counter > 0) {
+                this.activity_counter --;
+            }
+            this.is_counter_updated = true;
+            this.$('.o_notification_counter').text(this.activity_counter);
+            this.$el.toggleClass('o_no_notification', !this.activity_counter);
         }
     },
+    get_activity_data: function(){
+        var self = this;
+
+        return self._rpc({
+            model: 'mail.activity',
+            method: 'get_activities',
+        }).then(function (data) {
+            self.is_counter_updated = false;
+            self.activities = data;
+            self.activity_counter = _.reduce(data, function(total_count, p_data){ return total_count + p_data.total_count; }, 0);
```

Comment:
@tivisse ,  Get total activity counter to display on top systray I have used reduce function : http://underscorejs.org/#reduce Thanks